### DISCUSSION TITLE
assume a role before calling GetCredentials to redshift serverless

### DISCUSSION THREAD

**Discussion by mdeshmu** [aws/amazon-redshift-python-driver#170] [Q&A]
We have a Python application running on ECS and connecting to Redshift serverless using redshift_connector. By default, redshift_connector uses ECS task's IAM role to get serverless credentials. 
We want to specify an IAM role other than ECS task IAM role. 
Is it supported in redshift_connector to assume a role and then call getCredentials for redshift serverless? 
If yes, how? 
If no, can you please take it as a feature request? Thanks 

## Driver version
Redshift Python Driver 2.0.911

## Redshift Serverless version
PostgreSQL 8.0.2 on i686-pc-linux-gnu, compiled by GCC gcc (GCC) 3.4.2 20041017 (Red Hat 3.4.2-6.fc3), Redshift 1.0.51314

## Client Operating System
ECS Fargate 1.4.0

## Python version
3.8.16

**Comment by Brooke-white** [comment]
Hi @mdeshmu , this functionality is not currently supported. I will discuss this feature request with the team and report back. In the meantime, I'd recommend looking to utilize the boto3 API to assume the desired IAM role prior to invoking redshift-connector.

**Reply by mdeshmu** [reply]
Thanks for response. I will wait for your comment. 

IMO, It would be really useful feature (specially in BI tools). For example, PyAthena already supports this.

**Reply by mdeshmu** [reply]
Hi @Brooke-white, Waiting to hear about the internal discussion that you spoke about in your last comment.
